src/nix.rs

```diff
@@ -3,13 +3,37 @@ use crate::conversion::State;
 pub use crate::conversion::ToNickel;
 use crate::identifier::Ident;
 use crate::mk_app;
-use crate::parser::utils::mk_span;
+use crate::parser::utils::{mk_span, FieldPathElem};
 use crate::term::make::{self, if_then_else};
 use crate::term::{record::RecordData, BinaryOp, UnaryOp};
 use crate::term::{MergePriority, MetaValue, RichTerm, Term};
 use codespan::FileId;
-use rnix::ast::{BinOp as NixBinOp, Path as NixPath, Str as NixStr, UnaryOp as NixUniOp};
+use rnix::ast::{BinOp as NixBinOp, Path as NixPath, Str as NixStr, UnaryOp as NixUniOp, Attr as NixAttr};
 use std::collections::HashMap;
+use std::iter::repeat_with;
+
+fn path_elem_from_nix(attr: NixAttr, state: &State) -> FieldPathElem {
+    match attr {
+        NixAttr::Ident(id) => FieldPathElem::Ident(id.to_string().into()),
+        NixAttr::Str(s) => FieldPathElem::Expr(s.translate(state)),
+        NixAttr::Dynamic(d) => FieldPathElem::Expr(d.expr().unwrap().translate(state)),
+    }
+}
+
+fn path_elem_rt(attr: NixAttr, state: &State) -> RichTerm {
+    match attr {
+        NixAttr::Ident(id) => Term::Str(id.to_string()).into(),
+        NixAttr::Str(s) => s.translate(state),
+        NixAttr::Dynamic(d) => d.expr().unwrap().translate(state),
+    }
+}
+
+fn path_rts_from_nix<T>(n: rnix::ast::Attrpath, state: &State) -> T
+where
+T: FromIterator<RichTerm>
```

Comment:
It's strange to take a parameter `T` that only appears in the output (in fact I didn't know you could do that). I think writing an `impl` constraint is more idiomatic:
```suggestion
fn path_rts_from_nix<T>(n: rnix::ast::Attrpath, state: &State) -> impl FromIterator<RichTerm>
```

Reply:
We can drop the `<T>` in this case

Reply:
Actualy, using the `impl ...` syntax, rust complains that we have to know the type at this point.
The form `-> T where T: ...` probably work because the function is private.

Reply:
Oh, I missed the `collect()` at the end. Now it all makes sense, and should be a `<T>`. Sorry for the noise